src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -2757,6 +2776,357 @@ theorem get!_union_of_contains_eq_false_right [EquivBEq α] [LawfulHashable α]
 
 end Const
 
+section Inter
+
+variable {m₁ m₂ : Raw₀ α β}
+
+/- contains -/
+theorem contains_inter [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF)
+    (h₂ : m₂.val.WF) {k : α} :
+    (m₁.inter m₂).contains k = (m₁.contains k && m₂.contains k) := by
+  simp_to_model [contains, inter] using List.containsKey_filter_containsKey
+
+theorem contains_inter_iff [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF)
+    (h₂ : m₂.val.WF) {k : α} :
+    (m₁.inter m₂).contains k ↔ m₁.contains k ∧ m₂.contains k := by
+  simp_to_model [inter, contains] using List.containsKey_filter_containsKey_iff
+
+theorem not_contains_inter_of_not_contains_left [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) {k : α}
+    (h : m₁.contains k = false) :
+    (m₁.inter m₂).contains k = false := by
+  revert h
+  simp_to_model [inter, contains] using List.not_contains_filter_containsKey_of_not_containsKey_left
+
+theorem not_contains_inter_of_not_contains_right [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) {k : α}
+    (h : m₂.contains k = false) :
+    (m₁.inter m₂).contains k = false := by
+  revert h
+  simp_to_model [inter, contains] using List.not_contains_filter_containsKey_of_not_containsKey_right
+
+/- get? -/
+theorem get?_inter [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) {k : α} :
+    (m₁.inter m₂).get? k =
+    if m₂.contains k then m₁.get? k else none := by
+  simp_to_model [inter, get?, contains] using List.getValueCast?_filter_containsKey
+
+theorem get?_inter_of_contains_right [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₂.contains k) :
+    (m₁.inter m₂).get? k = m₁.get? k := by
+  revert h
+  simp_to_model [inter, get?, contains] using List.getValueCast?_filter_containsKey_of_containsKey_right
+
+theorem get?_inter_of_not_contains_left [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₁.contains k = false) :
+    (m₁.inter m₂).get? k = none := by
+  revert h
+  simp_to_model [inter, get?, contains] using List.getValueCast?_filter_containsKey_of_not_containsKey_left
+
+theorem get?_inter_of_not_contains_right [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₂.contains k = false) :
+    (m₁.inter m₂).get? k = none := by
+  revert h
+  simp_to_model [inter, get?, contains] using List.getValueCast?_filter_containsKey_of_not_containsKey_right
+
+/- get -/
+theorem get_inter [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} {h_contains : (m₁.inter m₂).contains k} :
+    (m₁.inter m₂).get k h_contains =
+    m₁.get k (((@contains_inter_iff α β _ _ m₁ m₂ _ _ h₁ h₂ k).1 h_contains).1) := by
```

Comment:
I am surprised by your fascination with `@`. Do you not like
```suggestion
    m₁.get k ((contains_inter_iff h₁ h₂).1 h_contains).1 := by
```
for some reason?